halo2_proofs/src/plonk/circuit.rs

```diff
@@ -478,6 +479,15 @@ impl Selector {
     }
 }
 
+/// Query of fixed column at a certain relative location
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct FixedQueryMid {
```

Comment:
The name `Mid` comes from `Middleware` :sweat_smile: , it was just the first thing that came to my mind.  The main point of the new names is to avoid collisions, so I added suffixes to some new types but I didn't think too much about it, I'm open to any suggestion!

Another point is that currently the state of the PR is that everything is still a single crate.  After splitting into crates some types will go do different types so possibly the names can be back to normal.  For example, instead of having `FixedQuery` and `FixedQueryMid` you would have `halo2_middleware::FixedQuery` and `halo2_backend::FixedQuery`.

`FixedQuery` includes a query index, which points to a vector of queries.  But in `FixedQueryMid` this vector doesn't exist (to remove the burden from the frontend of keeping track of queries).